### PR TITLE
[Issue #339] Nicer debug checks

### PR DESCRIPTION
**Pull request by colezemel**
Response is logged to console because error message isn't working properly.

**Unit test fails on purpose:**
Temporarily tests with status = failed for debugging (to see the functionality)

Is this the kind of function you were looking for?

### CODE REVIEW

File: frontend/src/tests/utils.js [hunk 1]
```diff
@@ -36,6 +36,25 @@ expect.extend({
...
+ * @param {object} resp The recieved array
+ * @param {object} obj  The json object to compare resp to
+ * @returns
+ */
+export function toMatchSuccessDebug(resp) {
+    try {
+        expect(resp).toMatchObject({ status: "failed" });
+    } catch (err) {
+        // err.message = "resp: \n", resp`;
+        console.log("resp: \n", resp);
```
Comment (siefkenj) [inline frontend/src/tests/utils.js]: You shouldn't use console.log. Look at how the proptypes check was done.

Comment (colezemel) [reply frontend/src/tests/utils.js]: Changed to console.warn

File: frontend/src/tests/utils.js [hunk 2]
```diff
@@ -36,6 +36,23 @@ expect.extend({
...
+ * @export
+ * @param {object} resp The recieved array
+ * @returns
+ */
+export function toMatchSuccessDebug(resp) {
+    try {
+        expect(resp).toMatchObject({ status: "success" });
+    } catch (err) {
+        console.warn("resp: \n", resp);
+        throw err;
```
Comment (siefkenj) [inline frontend/src/tests/utils.js:52]: This is still not right. The error message needs to be descriptive, and you shouldn't use console.warn.

File: frontend/src/tests/instructor-tests.js [hunk 3]
```diff
@@ -9,7 +9,8 @@ import {
     expect,
     it,
     beforeAll,
-    afterAll
+    afterAll,
```
Comment (siefkenj) [inline frontend/src/tests/instructor-tests.js:12]: Where is this used? It probably shouldn't be...

File: frontend/src/tests/utils.js [hunk 4]
```diff
@@ -36,6 +36,23 @@ expect.extend({
...
 
+/**
+ * Customize the `.toMatchObject` method to print the contents of the recieved
+ * object if the response call fails
+ *
+ * @export
+ * @param {object} resp The recieved array
+ * @returns
+ */
+export function toMatchSuccessDebug(resp) {
```
Comment (siefkenj) [inline frontend/src/tests/utils.js]: This isn't a good name. Come up with a better one :-)

File: frontend/src/tests/instructor-tests.js (hunk 3 above)
Comment (colezemel) [reply frontend/src/tests/instructor-tests.js:12]: I didn't change this

File: frontend/src/tests/utils.js (hunk 2 above)
Comment (colezemel) [reply frontend/src/tests/utils.js:52]: Ok think I finally got it. I was missing `JSON.stringify`

File: frontend/src/tests/instructor-tests.js (hunk 3 above)
Comment (siefkenj) [reply frontend/src/tests/instructor-tests.js:12]: Please look into it and file an issue/PR if needed.

File: frontend/src/tests/utils.js (hunk 2 above)
Comment (siefkenj) [reply frontend/src/tests/utils.js:52]: This is improved! But I think you can do better. Imagine you, in January, trying to work on your first bug. What error message would you hope to see? Write that message.

Also, the description of this function is not good. In particular, it doesn't explain why anyone should use this function.

**Review by CarolineYao — APPROVED** [review]

**Comment by siefkenj** [comment]
I merged the other refactoring, so there are a few conflicts now.

**Comment by siefkenj** [comment]
I've addressed part of this in #349

**Comment by colezemel** [comment]
#349 addressed this. I created #354 for the `afterAll` issue in Instructor

**Comment by colezemel** [comment]
#349 addressed this. I created #354 for the afterAll issue in Instructor

**Pull request closed without merging**